### PR TITLE
Adding FileSystemStorageDataSource

### PR DESCRIPTION
**Pull request by vilanovi**

### CODE REVIEW

File: Example/Harmony.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -480,7 +480,7 @@
 					};
 					607FACE41AFB9204008FA782 = {
 						CreatedOnToolsVersion = 6.3.1;
-						DevelopmentTeam = 3VP5WS89TZ;
+						DevelopmentTeam = 9YHGX7ST8H;
```
Comment (newboadki) [inline Example/Harmony.xcodeproj/project.pbxproj:483]: Is this change intentional?

File: Harmony/Classes/Repository/FileSystemStorageDataSource.swift [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    ///   - directory: The directory where to store data
+    init(fileManager: FileManager, directory: URL) {
+        self.fileManager = fileManager
+        self.directory = directory
+    }
+    
+    /// Convenience initializer. Returns nil if the document directory is not reachable.
+    ///
+    /// - Parameters:
+    ///   - fileManager: Teh FileManager
```
Comment (newboadki) [inline Harmony/Classes/Repository/FileSystemStorageDataSource.swift]: Typo.

**Review by newboadki — APPROVED** [review]

**Review by joselufo — APPROVED** [review]
all OK!

**Comment by franmontiel** [comment]
It can be a good idea not to use the key from the KeyQuery directly as there are characters that are invalid on paths. 
In my implementation on Android (pending to do a PR) I did a sha-256 hash of the key and this is used as the filename

**Comment by newboadki** [comment]
@vilanovi, please see Fran's comment and let us know your thoughts on the issue he presents, and whether you're planning on implementing the changes.

**Comment by vilanovi** [comment]
Good point @franmontiel. I agree with it. However, It's somehow nice to have the pair key - file, to make it simpler to debug or to access the file system in a human readable way.

What about adding a parameter by constructor of the data source that enables the key name hashing? or this is overengineered? This would help me make this change retrocompatible with the existing usages of this class in production projects, where no hash is being applied.

Thoughts?

**Comment by franmontiel** [comment]
For me it's ok to make it configurable by constructor. In case we want to have any default behavior I would say it should be with the hash option enabled (to avoid possible error when using it)

**Comment by vilanovi** [comment]
I'm adding a couple of extra changes and ping you for final review ASAP.

**Comment by vilanovi** [comment]
Added sha256 support! (by default). User can customize it to md5, custom file name encoding or none.
